api/admin/controller.py

```diff
@@ -1912,10 +1995,12 @@ def collections(self):
                 )
                 if c.protocol in [p.get("name") for p in protocols]:
                     [protocol] = [p for p in protocols if p.get("name") == c.protocol]
-                    libraries = [
-                            self._get_integration_library_info(
-                                c.external_integration, library, protocol)
-                            for library in c.libraries]
+                    libraries = []
+                    for library in c.libraries:
+                        if not flask.request.admin or not flask.request.admin.is_librarian(library):
```

Comment:
When I was reviewing the core branch, I didn't mind too much that there was no caching of AdminRole objects, because I figured permissions were only being checked once per request. But this code gives me pause. If we have thirty libraries that share five collections (which I think is reasonable) one call to `collections will make 5*30*2=300 calls to `is_librarian`, and each `is_librarian` call might make 3 SQL queries.

Can you change it so looking up an `Admin` object preloads all their `AdminRoles`? Making `AdminRole` a `HasFullTableCache` would also basically solve this problem.